README.md

```diff
@@ -6,106 +8,295 @@
+functionality. See [To-dos](#to-dos) below.
+
+## Qless Features
+
+For a more complete discussion of the features, consult the documentation of the
+[Python bindings](http://github.com/seomoz/qless-py). The short list:
+
+- Recurring jobs
+- Job heartbeating / exclusive locks
+- Retries for dropped jobs
+- Job dependencies (useful for creating chains of jobs)
+- Scheduled jobs
+- __TODO__: Job sandboxes
+
+## Installation
+
+Available on `npm`:
+
+```bash
+npm install qless-js
+```
+
+## Business Time
+
+All asychronous parts of the API (which is most of it), return promises.
+
+This example puts a job of class `mymodule.MyClass` in a queue `myQueue` in a
+local redis instance:
 
-### Example
-#### Example enqueuer
 ```javascript
 // myproject/enqueue.js
 'use strict';
+
+// Create our client
 const qless = require('qless');
-const client = new qless.Client();
-client.queue('myqueue').put('MyClass', {foo: 'bar'}, {}, (err, res) => {
-  if (err) {
-    console.log("error: ", err, err.message);
-  } else {
-    console.log('success');
-    process.exit(0);
-  }
-});
+const client = new qless.Client('redis://localhost:6379')
+
+// Pick a queue
+let queue = client.queue('myQueue');
+
+// Put a job in that queue
+let jobConfig = {
+  klass: 'mymodule.MyClass',
+  data: {
+    foo: 'bar',
+  },
+}
+queue.put(jobConfig).then((jid) => {
+  console.log('Put jid', jid);
+}).catch((err) => {
+  console.log('Exception', err);
+})
 ```
 
-#### Example worker
+But when this job is run, what will it do? To this end, let's define our class.
+There are more ways than this to define our jobs, but more on that in the
+[Job Class Importing](#job-class-importing) section below.
+
+The function that gets run is `<job class>.<queue name>`. So our example will
+trun a `MyClass.myQueue` function, passing the popped job. This function:
+
+- _must_ return a promise
+- _should_ ultimately return the `job.complete()` promise if successful
+
+When processing the job, any exceptions propagated out will result in the job
+being marked as `failed`.
+
 ```javascript
-// myproject/worker.js
-'use strict';
+// mymodule
+class MyClass {
 
-const qless = require('qless');
-qless.klassFinder.setModuleDir(__dirname + '/jobs');
-const client = new qless.Client();
-const worker = new qless.SerialWorker('myqueue', client);
-
-worker.run(err => {
-  console.log("ERROR IN WORKER: ", err);
-  // normally won't happen unless a serious (e.g. redis) error
-  // NOT triggered when a job [safely] fails
-});
+  // This method has two requirements:
+
+  static myQueue(job) {
+    return someAsynchronousWork()
+      .then(() => {
+        return someMoreAsynchronousWork();
+      })
+      .then(() => {
+        return job.complete();
+      })
+  }
+}
 ```
 
-To enable debugging, run with:
+As for running these jobs, `qless-js` comes with a binary that will continuously
+pop and run jobs:
 
 ```bash
-DEBUG='qless:*' node worker.js
+qless-js-worker \
```

Comment:
WRT installation, my expectation is that it would be installed in the environment in which it's deployed. Be that globally or `yarn`